source/class.cpp

```diff
@@ -1172,12 +1184,23 @@ void ClassBinder::bind(Context &context)
 	string const trampoline_name = callback_structure_constructible ? callback_structure_name(C) : "";
 	string const binding_qualified_name = callback_structure_constructible ? callback_structure_name(C) : qualified_name;
 
-	string maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name); // for now enable std::shared_ptr by default
-	if( is_inherited_from_enable_shared_from_this(C) ) maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name);
+	string shared_ptr = "std::shared_ptr";
+	bool use_custom_shared = Config::get().use_custom_shared();
+	if( use_custom_shared )
+		shared_ptr = Config::get().custom_shared();
+
+
+	string maybe_holder_type = ", {}<{}>"_format(shared_ptr, qualified_name);
+	//Check if the type is a custom shared pointer:
+	if( use_custom_shared && qualified_name.rfind(shared_ptr, 0) == 0 ) {
+		maybe_holder_type = "";
+	}
```

Comment:
Again, could you please elaborate whats going on here? Thanks,

Reply:
Same as above and same comment w.r.t https://github.com/RosettaCommons/binder/blob/master/source/type.cpp#L662